sqlite3_opt_serialize_test.go

```diff
@@ -0,0 +1,101 @@
+// +build !libsqlite3 sqlite_serialize
+
+package sqlite3
+
+import (
+	"database/sql"
+	"os"
+	"testing"
+)
+
+func TestSerializeDeserialize(t *testing.T) {
+	// The driver's connection will be needed in order to serialization and deserialization
+	driverName := "TestSerializeDeserialize"
+	driverConns := []*SQLiteConn{}
+	sql.Register(driverName, &SQLiteDriver{
+		ConnectHook: func(conn *SQLiteConn) error {
+			driverConns = append(driverConns, conn)
```

Comment:
As I mentioned, this is not a safe way to do this. In practice, developers should use the `Raw` method like so:
```go
srcDb, err := sql.Open("sqlite3", srcTempFilename)
if err != nil {
    t.Fatal(err)
}
defer srcDb.Close()

// insert into src
...

srcConn, err := srcDb.Conn(context.Background())
if err != nil {
    t.Fatal(err)
}

var serialized []byte
if err := srcConn.Raw(func(raw interface{}) error {
    var err error
    serialized, err = raw.(*sqlite3.SQLiteConn).Serialize("")
    return err
}); err != nil {
    t.Fatal(err)
}

srcConn.Close()

destDb, err := sql.Open("sqlite3", destTempFilename)
if err != nil {
    t.Fatal(err)
}
defer destDb.Close()

destConn, err := destDb.Conn(context.Background())
if err != nil {
    t.Fatal(err)
}

if err := destConn.Raw(func(raw interface{}) error {
    return raw.(*sqlite3.SQLiteConn).Deserialize(serialized, "")
}); err != nil {
    t.Fatal(err)
}

destConn.Close()

// now query destDb to verify all the data is present
...
```